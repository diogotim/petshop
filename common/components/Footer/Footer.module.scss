@import '../../styles/mixins';

.root {
    margin-top: 1rem;
}

.footer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    gap: 4rem;
}

.logo {
    margin: auto;
}

@include md {
    .logo {
        margin: 0;
    }
}

.nav {
    ul {
        list-style-type: none;
        display: flex;
        flex-wrap: wrap;
        gap: 8rem;
    }

    p {
        font-weight: 900;
        font-family: var(--font-display);
    }

    a {
        display: block;
    }

    a:hover {
        color: var(--main);
    }
}