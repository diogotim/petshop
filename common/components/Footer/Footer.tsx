import React from "react";
import Link from "next/link";
import classes from "./Footer.module.scss";

const Footer: React.FunctionComponent = () => {
  return (
    <footer className={classes.root}>
      <div className={`${classes.footer} container`}>
        <Link href="/">
          <a className={classes.logo}>Pets</a>
        </Link>
        <nav className={classes.nav}>
          <ul>
            <li>
              <p>Quick links</p>
              <Link href="/">Adoption</Link>
              <Link href="/">Services</Link>
              <Link href="/products/1">Products</Link>
            </li>
            <li>
              <p>Follow us</p>
              <a href="#">Facebook</a>
              <a href="#">Instagram</a>
              <a href="#">Twitter</a>
            </li>
            <li>
              <p>Contact</p>
              <a href="#">+0 (000) 000-0000</a>
              <a href="#">someone@example.com</a>
            </li>
          </ul>
        </nav>
      </div>
    </footer>
  );
};

export default Footer;
