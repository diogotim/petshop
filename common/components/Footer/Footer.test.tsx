import { render } from "@testing-library/react";
import Footer from "./Footer";

describe("Footer", () => {
  it("renders footer unchanged", () => {
    const footer = render(<Footer />);
    expect(footer).toMatchSnapshot();
  });

  it("renders a contentinfo", () => {
    const { getAllByRole } = render(<Footer />);
    expect(getAllByRole("contentinfo").length).toBe(1);
  });
});
