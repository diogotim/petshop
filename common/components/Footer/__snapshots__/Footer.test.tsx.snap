// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footer unchanged 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="root"
      >
        <div
          class="footer container"
        >
          <a
            class="logo"
            href="/"
          >
            Pets
          </a>
          <nav
            class="nav"
          >
            <ul>
              <li>
                <p>
                  Quick links
                </p>
                <a
                  href="/"
                >
                  Adoption
                </a>
                <a
                  href="/"
                >
                  Services
                </a>
                <a
                  href="/products/1"
                >
                  Products
                </a>
              </li>
              <li>
                <p>
                  Follow us
                </p>
                <a
                  href="#"
                >
                  Facebook
                </a>
                <a
                  href="#"
                >
                  Instagram
                </a>
                <a
                  href="#"
                >
                  Twitter
                </a>
              </li>
              <li>
                <p>
                  Contact
                </p>
                <a
                  href="#"
                >
                  +0 (000) 000-0000
                </a>
                <a
                  href="#"
                >
                  someone@example.com
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="root"
    >
      <div
        class="footer container"
      >
        <a
          class="logo"
          href="/"
        >
          Pets
        </a>
        <nav
          class="nav"
        >
          <ul>
            <li>
              <p>
                Quick links
              </p>
              <a
                href="/"
              >
                Adoption
              </a>
              <a
                href="/"
              >
                Services
              </a>
              <a
                href="/products/1"
              >
                Products
              </a>
            </li>
            <li>
              <p>
                Follow us
              </p>
              <a
                href="#"
              >
                Facebook
              </a>
              <a
                href="#"
              >
                Instagram
              </a>
              <a
                href="#"
              >
                Twitter
              </a>
            </li>
            <li>
              <p>
                Contact
              </p>
              <a
                href="#"
              >
                +0 (000) 000-0000
              </a>
              <a
                href="#"
              >
                someone@example.com
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
